### ISSUE TITLE
Missing semanticUI 'required' class on ProvinceCode Select Input

### ISSUE THREAD

**Issue by acornforth** [Sylius/Sylius#12681]
**Sylius version affected**: 1.6 - 1.9 (as far as i can tell)

**Description**  
When the onChange event on the country selector triggers the province field to load as a SELECT element, it is a required field, but the 'required' class is not set on the parent element and as a result, the red asterisk provided by that semanticUI class is not displayed.

**Steps to reproduce**  

Having a country set up with provinces, reach the addressing step in checkout, and select the country with provinces from the country select element. Observe that red asterisk does not appear on the province dropdown, even though the SELECT itself has the 'required' attribute.

**Possible Solution**  
 add
```
provinceContainer.addClass('required');
```
 to Line 64 of `src/Sylius/Bundle/ShopBundle/Resources/private/js/sylius-province-field.js` 
 
 with corresponding removeClass calls in the other branches of the if-else structure.

I can make a PR for this if someone can advise me which repo/branch to submit it to, i notice that ShopBundle is a readonly repo. 

I don't think the fix would be a breaking change, and it's a bit of a pain to override private js assets from these Bundles, so would be nice to see a fix rolled out in sylius itself.

**Comment by acornforth** [comment]
Anyone looking for a way to workaround this you can just add
```sass
.province-container.field > label.required:after
  display: inline-block
  vertical-align: top
  margin: -0.2em 0em 0em 0.2em
  content: '*'
```
to your styles somewhere.

**Comment by NoResponseMate** [comment]
Hello @acornforth 
We can confirm the issue, and your proposed fix should do the trick.
Could you open a PR with it?

**Comment by Wojdylak** [comment]
I am closing this issue as it has been resolved by PR #15900
